src/context/func.rs

```diff
@@ -154,6 +154,16 @@ pub trait FuncCaller: GraphLike + AnalyzerLike<Expr = Expression> + Sized {
                             let cvar = self.add_node(Node::ContextVar(var));
                             ExprRet::Single((ctx, cvar))
                         }
+                        "keccak256" => {
+                            self.parse_ctx_expr(&input_exprs[0], ctx).expect_single();
+                            let var = ContextVar::new_from_builtin(
```

Comment:
If the input is `if let VarType::Concrete(cnode) = cvar.underlying(self).ty`, we can actually do the keccak256 at analysis time, and otherwise we do the `new_from_builtin` call.

This would look like: 
```rust
if let VarType::Concrete(cnode) = cvar.underlying(self).ty {
    let conc = cnode.underlying(self);
    let hashed = conc.keccak256();
    let new_conc = self.add_node(Node::ContextVar(ContextVar::new_from_concrete(hashed)));
    ExprRet::Single((ctx, new_conc))
} else {
    // add builtin
}
```

and inside `nodes/concrete.rs`, add a `keccak256(&self) -> Concrete` function that hashes the concrete variable. In the future, we should add support for static user types as well (i.e. static structs) but lets punt on that